gen/go/eventschema/getters.go

```diff
@@ -0,0 +1,156 @@
+	fields := events2.EventFields{"event": eventType}
+	eventStruct, err := events2.FromEventFields(fields)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	messageName := getMessageName(eventStruct)
+	event, ok := events[messageName]
+	if !ok {
+		return nil, trace.NotFound("message %s unknown", messageName)
+	}
+	return event, nil
+}
+
+func getMessageName(eventStruct interface{}) string {
+	if t := reflect.TypeOf(eventStruct); t.Kind() == reflect.Ptr {
+		return t.Elem().Name()
+	} else {
+		return t.Name()
+	}
+}
+
+func TableList() (string, error) {
+	sb := strings.Builder{}
+	sb.WriteString("table_name, description\n")
+	for _, name := range eventTypes {
+		eventSchema, err := GetEventSchemaFromType(name)
+		if err != nil {
+			return "", trace.Wrap(err)
+		}
+		sb.WriteString(fmt.Sprintf("%s, %s\n", name, eventSchema.Description))
+	}
+	return sb.String(), nil
+}
+
+func (event *Event) TableSchema() (string, error) {
+	sb := strings.Builder{}
+	sb.WriteString("column_name, column_type, description\n")
+	for propName, prop := range event.Fields {
+		line, err := prop.TableSchema(propName)
+		if err != nil {
+			return "", trace.Wrap(err)
+		}
+		sb.WriteString(line)
+	}
+	return sb.String(), nil
+}
+
+func (field *EventField) TableSchema(name string) (string, error) {
+	sb := strings.Builder{}
+	switch field.Type {
+	case "object":
+		for propName, prop := range field.Fields {
+			line, err := prop.TableSchema(fmt.Sprintf("%s_%s", name, propName))
+			if err != nil {
+				return "", trace.Wrap(err)
+			}
+			sb.WriteString(line)
+		}
+	case "string", "integer", "boolean", "date-time":
+		sb.WriteString(tableSchemaLine(name, dmlType[field.Type], field.Description))
+	case "array":
+		sb.WriteString(tableSchemaLine(name, fmt.Sprintf("array(%s)", dmlType[field.Items.Type]), field.Description))
+	default:
+		return "", trace.NotImplemented("field type '%s' not supported", field.Type)
+	}
+	return sb.String(), nil
+}
+
+func (event *Event) ViewSchema() (string, error) {
+	sb := strings.Builder{}
+	sb.WriteString("SELECT\n")
+	sb.WriteString("  event_date, event_time,\n")
+	for propName, prop := range event.Fields {
+		line, err := prop.ViewSchema([]string{propName})
+		if err != nil {
+			return "", trace.Wrap(err)
+		}
+		sb.WriteString(line)
+	}
+	return sb.String(), nil
+}
+
+func (field *EventField) ViewSchema(path []string) (string, error) {
+	sb := strings.Builder{}
+	switch field.Type {
+	case "object":
+		for propName, prop := range field.Fields {
+			line, err := prop.ViewSchema(append(path, propName))
+			if err != nil {
+				return "", trace.Wrap(err)
+			}
+			sb.WriteString(line)
+		}
+	case "string", "integer", "boolean", "date-time":
+		sb.WriteString(viewSchemaLine(jsonFieldName(path), path[len(path)-1], dmlType[field.Type]))
+	case "array":
+		sb.WriteString(viewSchemaLine(jsonFieldName(path), path[len(path)-1], fmt.Sprintf("array(%s)", dmlType[field.Items.Type])))
```

Comment:
In case of AccessRequestCreate.resource_ids where resource_ids is defined as Object type without underlying `dmlType`  mapping the `array(%s)`  we will generate the `'$["resource_ids"]') AS array(varch) as resource_ids` entry that is invalid. 

Can you run all the `ViewSchema` against Athena SQL to check the correctness of View Schema ?

Reply:
I fixed the view generation to support array of objects and fields with `.` in them. Also the comma moved to the beginning of the line to avoid having to remove the comma on the last line and be consistent with how the Athena Query builder exports view queries.

I manually ran all the views through Athena and none of them error. I have not teststed querying them as I don't have real data yet.

I will add unit tests covering each dmlType.